test/unit/util/test_custom_templates.py

```diff
@@ -0,0 +1,59 @@
+"""Test custom template configuration and rendering."""
+
+from galaxy.util.custom_templates import template
+
+TEMPLATE_RELPATH = "mail/activation-email.txt"
+CONTEXT = {
+    "name": "Jane Doe",
+    "user_email": "jane@example.com",
+    "date": "01-01-2021",
+    "hostname": "example.galaxy.org",
+    "activation_url": "https://example.galaxy.org/fake-activation-link",
+    "terms_url": "https://example.galaxy.org/terms",
+    "contact_email": "contact@example.galaxy.org",
+    "instance_resource_url": "https://example.galaxy.org/resources",
+    "expiry_days": 7,
+}
+CUSTOM_TEMPLATE = """
+Here's my custom template. It starts with a comment section to let admins
+know how to use it.
+
+>>>>>> The actual template starts from the next line
+This is my custom template!
+
+Name: {{ name }}
+
+"""
+DEFAULT_TEMPLATE_HEADER_SUBSTRING = "The following variables are available"
+DEFAULT_TEMPLATE_OUTPUT_SUBSTRING = f"Hello {CONTEXT['name']}"
+CUSTOM_TEMPLATE_OUTPUT_SUBSTRING = f"Name: {CONTEXT['name']}"
+
+
+def test_it_can_render_a_default_template(tmp_path):
+    """It should fall back on codebase default."""
+    custom_templates_dir = tmp_path
+    output = template.render(
+        TEMPLATE_RELPATH,
+        CONTEXT,
+        custom_templates_dir,
+    )
+    # Make sure it's parsed out the doc header
+    assert DEFAULT_TEMPLATE_HEADER_SUBSTRING not in output
```

Comment:
This is fine. However, if that line is modified in the default template (which is not impossible), this assertion won't tell us anything (i.e., assert that a line that is not in the template is not in the rendered template). I think a tighter check would be to verify against something that *should be* rendered and is unlikely to be modified. For example, if you take the html template, it starts with the `!DOCTYPE` tag - that's unlikely to be dropped. So you could check that the rendered template *starts with that tag*: that ensures that anything preceding that tag (which is the comment block) is not rendered.

Reply:
Done! Thanks a lot for the thorough review John :grin: